src/Hedgehog/Property.fs

```diff
@@ -81,11 +81,32 @@ module Property =
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
         bindGen (toGen << k) (toGen m) |> ofGen
 
+#if !FABLE_COMPILER
+    let private (|IsGenericList|_|) (candidate : obj) : seq<obj> option =
+        let t = candidate.GetType()
+        // have to use TypeInfo due to targeting netstandard 1.6
+        let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
+        let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
+        if isList
+        then Some (candidate :?> System.Collections.IEnumerable |> Seq.cast<obj>)
+        else None
+
+    let private printValue (value) : string =
+        match value with
+        | IsGenericList list ->
+            let printedElements = Seq.map (fun element -> sprintf "%A" element) list
+            "[" + (String.concat ", " printedElements) + "]"
```

Comment:
How about mapping the value `list` to be of type `obj list`?  Then no additional "to string" code is needed.

Then I expect this function could be renamed to something like `improveRuntimeType` and the implementation below would be
```fsharp
x |> improveRuntimeType |> sprintf "%A"
```

Reply:
I tried to convert the active pattern to a regular function but it did not typecheck (obj vs obj list)
Otherwise - done

Reply:
I like the use of an active pattern here.  I am not suggesting removing it.